### DISCUSSION TITLE
Redirect after sign in with GitHub OAuth

### DISCUSSION THREAD

**Discussion by infamous55** [supabase/supabase#15093] [Questions] [answered]
Hello! I'm currently using Next.js 13.4 with the app router and the `@supabase/auth-helpers-nextjs` package.

I would like to redirect the user to `/dashboard` from a client component called `NavBar` after they log in using the GitHub OAuth provider.

I attempted to use the `redirectTo` option with a value of **"http://localhost:3000/dashboard"** when calling `signInWithOAuth`, but it doesn't seem to work. I made sure to include the redirect URL in the list of authorized redirect URLs in the dashboard.

Even manually redirecting with `router.push()` doesn't work (possibly due to [this issue](https://github.com/supabase/auth-helpers/issues/578)).

Here's how my code looks:

```ts
const handleSignIn = async () => {
    try {
        const { error } = await supabase.auth.signInWithOAuth({
            provider: 'github',
            options: {
                scopes: 'repo',
                // redirectTo: 'http://localhost:3000/dashboard',
            },
        });

        if (error) throw error;

        // router.push('/dashboard');
    } catch {
        toast.error('Sign in failed!');
    }
};
```

I'm posting this as a question because I'm uncertain whether I misunderstood something, if I'm missing some information, or if there might be a bug involved.

Any suggestions or insights on how to achieve the desired behavior would be great. Thanks!

**Comment by tholder** [comment, marked as answer]
You should migrate to PKCE auth flow now. See https://supabase.com/docs/guides/auth/auth-helpers/nextjs#migrating-to-v07x

We have an auth/callback/route.ts file now that looks like

```typescript
import { createRouteHandlerClient } from '@supabase/auth-helpers-nextjs'
import { cookies } from 'next/headers'
import { NextResponse } from 'next/server'

import type { NextRequest } from 'next/server'
import type { Database } from '@src/database.types'

export async function GET(request: NextRequest) {
  const requestUrl = new URL(request.url)
  const code = requestUrl.searchParams.get('code')
  const redirectPath = requestUrl.searchParams.get('redirect')
  const redirect = process.env.NEXT_PUBLIC_APP_URL! + redirectPath

  if (code) {
    const supabase = createRouteHandlerClient<Database>({ cookies })
    await supabase.auth.exchangeCodeForSession(code)
  }

  // URL to redirect to after sign in process completes
  return NextResponse.redirect(redirect ? redirect : requestUrl.origin)
}

**Reply by tholder** [reply]
Our auth looks like:

```typescript
const redirectTo = getLoginRedirectPath(pathname)

    const { error } = await supabase.auth.signInWithOAuth({
      provider,
      options: {
        redirectTo,
        queryParams: {
          prompt: 'consent',
        },
      },
    })
```

```typescript
export const getLoginRedirectPath = (pathname?: string | null): string => {
  return (
    process.env.NEXT_PUBLIC_APP_URL +
    '/auth/callback' + // Required for PKCE authentication.
    '?redirect=' + // Passed to auth/route/callback to redirect after auth
    (pathname ? pathname : '/dashboard')
  )
}
```

**Reply by infamous55** [reply]
Thank you so much for your quick response - I appreciate it! 😄

**Comment by 26medias** [comment]
@infamous55 You use `scopes: 'repo'`, how do you obtain the github token to actually call the github api and list those repos?

**Reply by infamous55** [reply]
Hello @26medias! I simply requested the user to generate a fine-grained personal access token and provide it to the app I was working on.
